File: Leetcode/RunningSumOf1dArray.py
# Given an array nums. We define a running sum of an array as runningSum[i] = sum(nums[0]…nums[i]).
# Return the running sum of nums.

class Solution(object):
    def runningSum(self, nums):
        result = [nums[0]]
        
        for i in range(1, len(nums)):
            result.append(result[-1] + nums[i])
            
        return result

File: Leetcode/RunningSumOf1dArray.cs
// Given an array nums. We define a running sum of an array as runningSum[i] = sum(nums[0]…nums[i]).
// Return the running sum of nums.

public class Solution 
{
    public int[] RunningSum(int[] nums) 
    {
        var output = new int[nums.Length];
        output[0] = nums[0];
        
        for (int i = 1; i < nums.Length; i++)
            output[i] = output[i - 1] + nums[i];
        
        return output;
    }
}